DuckDuckGo/HomeMessageViewSectionRenderer.swift

```diff
@@ -119,7 +119,16 @@ class HomeMessageViewSectionRenderer: NSObject, HomeViewSectionRenderer {
                 params
             }
         case .remoteMessage(let remoteMessage):
+            let onDidAppear = { [weak self] in
+                self?.homePageConfiguration.didAppear(message)
+            }
+
+            // call didAppear here to support marking messages as shown when they appear on the new tab page
+            // as a result of refreshing a config while the user was on a new tab page already.
+            onDidAppear()
```

Comment:
It's fishy, but it works. That said, a remote message appearing on user's new tab page is a rare situation (normally they will appear at app startup after being refreshed in background).